react/content-list.jsx

```diff
@@ -45,14 +58,30 @@ let ContentList = React.createClass({
     return this.setStateFromDattCore()
   },
 
+  componentWillUnmount: function () {
+    this.unmonitorDattCore()
+  },
+
   componentWillReceiveProps: function () {
     return this.setStateFromDattCore()
   },
 
   monitorDattCore: function () {
     let dattcore = this.props.dattcore
+    let uiEvents = this.props.uiEvents
+
     dattcore.on('peers-content-auth', this.handlePeersContentAuth)
     dattcore.on('content-content-auth', this.handleContentContentAuth)
+    uiEvents.on('refresh-content', this.handleRefreshContent)
+  },
+
+  unmonitorDattCore: function () {
```

Comment:
I see. Then that's fine for now, but this approach is prone to error. It would be really easy to add a listener and forget to add the unlistener. Probably the way this should really be handled (not necessarily right now) is for the UI to have one high-level place that does all the listening to dattcore events, which then filter down to all the components. Individual components would then not listen to dattcore events and not even use dattcore directly.